util/types/convert.go

```diff
@@ -138,6 +138,25 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
 		}
 		// TODO: check Flen
 		return x, nil
+	case mysql.TypeYear:
+		var (
+			intVal int64
+			err    error
+		)
+		switch x := val.(type) {
+		case string:
+			intVal, err = StrToInt(x)
+		default:
```

Comment:
mysql> insert into t select cast("2015-11-11 11:11:11" as datetime);
Query OK, 1 row affected (0.01 sec)
Records: 1  Duplicates: 0  Warnings: 0

I think you should support time type separately.  ToInt64 will not work for time type.